extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -222,6 +227,37 @@ const BLOCK_SRA_COMBINERS_ = [
   },
 ];
 
+/**
+ * Used to manage messages for different fluid ad slots.
+ *
+ * Maps a sentinel value to an object consisting of the impl to which that
+ * sentinel value belongs and the corresponding message handler for that impl.
+ * @type{!Object<string, !{instance: !AmpAdNetworkDoubleclickImpl, connectionEstablished: boolean}>}
+ */
+const fluidListeners = {};
+
+/**
+ * @param {!Event} event
+ * @private
+ */
+function fluidMessageListener_(event) {
+  const data = tryParseJson(getData(event));
+  if (event.origin != SAFEFRAME_ORIGIN || !data || !data['sentinel']) {
+    return;
+  }
+  const listener = fluidListeners[data['sentinel']];
+  dev().assert(listener, 'postMessage listener does not exist');
```

Comment:
We should probably make this more than a dev assert otherwise it would get compiled away and it entirely possible postmessages may include sentinel in an unintentional way.  Move this to the if clause.

Reply:
Done.